### PR TITLE
Feature/Use Zitadel Postgres Integration by default

### PR DESCRIPTION
**Pull request by r0b2g1t**
This PR replaces cockroachDB as default DB for Zitadel  in the getting started script to deploy script. Users can switch back to cockroachDB by setting the environment variable ```ZITADEL_DATABASE``` to ```cockroach```.

The PR fix some typos as well.

## Issue ticket number and link
- #1785 

### Checklist
- [x] Is it a bug fix
- [x] Is a feature enhancement

### CODE REVIEW

**Comment by mlsmaycon** [comment]
Awesome PR, @r0b2g1t; thanks. We actually had in our roadmap to make Postgres the default for the installer. Can you update the following check to reflect that?

```shell
  if [[ $ZITADEL_DATABASE == "" ]]; then
    CRDB=$(renderDockerComposeCockroachDB)
    ZITADEL_DB_ENV=$(renderZidatelCockroachDBEnv)
  elif [[ $DATABASE == "postgres" ]]; then
    CRDB=$(renderDockerComposePostgres)
    ZITADEL_DB_ENV=$(renderZidatelPostgresEnv)
  fi
```

**Comment by r0b2g1t** [comment]
@mlsmaycon thanks for the remark. I have set Postgres as default and added echos for the user about the usage of the environment variable.

File: infrastructure_files/getting-started-with-zitadel.sh [hunk 1]
```diff
@@ -461,6 +463,17 @@ initEnvironment() {
...
+    echo "Use Postgres as default Zitadel database."
+    echo "For using CockroachDB please the environment variable 'export ZITADEL_DATABASE=cockroach'."
+    CRDB=$(renderDockerComposePostgres)
+    ZITADEL_DB_ENV=$(renderZidatelPostgresEnv)
+  elif [[ $DATABASE == "cockroach" ]]; then
+    echo "Use CockroachDB as Zitadel database."
+    CRDB=$(renderDockerComposeCockroachDB)
+    ZITADEL_DB_ENV=$(renderZidatelCockroachDBEnv)
+  fi
+
```
Comment (mlsmaycon) [inline infrastructure_files/getting-started-with-zitadel.sh]:
```suggestion
  if [[ "$ZITADEL_DATABASE-x" == "cockroach-x" ]]; then
    echo "Use CockroachDB as Zitadel database."
    CRDB=$(renderDockerComposeCockroachDB)
    ZITADEL_DB_ENV=$(renderZidatelCockroachDBEnv)
  else  
    echo "Use Postgres as default Zitadel database."
    echo "For using CockroachDB please the environment variable 'export ZITADEL_DATABASE=cockroach'."
    CRDB=$(renderDockerComposePostgres)
    ZITADEL_DB_ENV=$(renderZidatelPostgresEnv)
  fi



```

Comment (mlsmaycon) [reply infrastructure_files/getting-started-with-zitadel.sh]: would also rename CRDB to ZDB

File: infrastructure_files/getting-started-with-zitadel.sh [hunk 2]
```diff
@@ -760,7 +793,7 @@ services:
   zitadel:
     restart: 'always'
     networks: [netbird]
-    image: 'ghcr.io/zitadel/zitadel:v2.31.3'
+    image: 'ghcr.io/zitadel/zitadel:v2.54.3'
```
Comment (mlsmaycon) [inline infrastructure_files/getting-started-with-zitadel.sh]: would update the dependency setting here:

```
    depends_on:
      zdb:
```

File: infrastructure_files/getting-started-with-zitadel.sh [hunk 3]
```diff
@@ -770,11 +803,24 @@ services:
...
+  netbird_zitadel_certs:
+
+networks:
+  netbird:
+EOF
+}
+
+renderDockerComposeCockroachDB() {
+  cat <<EOF
+  # CockroachDB for Zitadel
```
Comment (mlsmaycon) [inline infrastructure_files/getting-started-with-zitadel.sh:829]: service name from crdb to zdb

File: infrastructure_files/getting-started-with-zitadel.sh [hunk 4]
```diff
@@ -788,14 +834,32 @@ services:
...
-  netbird_zitadel_certs:
+EOF
+}
 
-networks:
-  netbird:
+renderDockerComposePostgres() {
+  cat <<EOF
+  # Postgres for Zitadel
+  crdb:
```
Comment (mlsmaycon) [inline infrastructure_files/getting-started-with-zitadel.sh]: rename service to zdb

File: infrastructure_files/getting-started-with-zitadel.sh [hunk 5]
```diff
@@ -788,14 +834,32 @@ services:
...
+    restart: 'always'
+    networks: [netbird]
+    image: 'postgres:16-alpine'
+    environment:
+      - POSTGRES_USER=root
+      - POSTGRES_PASSWORD=postgres
```
Comment (mlsmaycon) [inline infrastructure_files/getting-started-with-zitadel.sh]: if you want to configure a random password will be better. Maybe make it persistent using a file like the temp machine key

File: infrastructure_files/getting-started-with-zitadel.sh [hunk 6]
```diff
@@ -689,15 +717,20 @@ ZITADEL_DATABASE_COCKROACH_ADMIN_SSL_MODE=verify-full
...
+EOF
+}
+
+renderZidatelPostgresEnv() {
+  cat <<EOF
+ZITADEL_DATABASE_POSTGRES_HOST=crdb
+ZITADEL_DATABASE_POSTGRES_PORT=5432
+ZITADEL_DATABASE_POSTGRES_DATABASE=zitadel
+ZITADEL_DATABASE_POSTGRES_USER_USERNAME=zitadel
+ZITADEL_DATABASE_POSTGRES_USER_PASSWORD=zitadel
```
Comment (mlsmaycon) [inline infrastructure_files/getting-started-with-zitadel.sh]: Using a random password is safer.

**Review by mlsmaycon — CHANGES_REQUESTED** [review]
Thanks @r0b2g1t .

Can you also update the https://github.com/netbirdio/netbird/blob/main/.github/workflows/test-infrastructure-files.yml#L172 file with cockroachdb tests?

**Comment by r0b2g1t** [comment]
> Thanks @r0b2g1t .
> 
> Can you also update the https://github.com/netbirdio/netbird/blob/main/.github/workflows/test-infrastructure-files.yml#L172 file with cockroachdb tests?

I will take a look into it and will try to create tests. :-)

**Comment by r0b2g1t** [comment]
> > Thanks @r0b2g1t .
> > Can you also update the https://github.com/netbirdio/netbird/blob/main/.github/workflows/test-infrastructure-files.yml#L172 file with cockroachdb tests?
> 
> I will take a look into it and will try to create tests. 🙂

@mlsmaycon I have added tests for psotgres and cockroach.

File: infrastructure_files/getting-started-with-zitadel.sh [hunk 7]
```diff
@@ -461,6 +463,20 @@ initEnvironment() {
...
 
+  if [[ $ZITADEL_DATABASE == "" ]]; then
+    echo "Use Postgres as default Zitadel database."
+    echo "For using CockroachDB please the environment variable 'export ZITADEL_DATABASE=cockroach'."
+    POSTGRES_ROOT_PASSWORD="$(openssl rand -base64 32 | sed 's/=//g')@"
+    POSTGRES_ZITADEL_PASSWORD="$(openssl rand -base64 32 | sed 's/=//g')@"
+    ZDB=$(renderDockerComposePostgres)
+    ZITADEL_DB_ENV=$(renderZitadelPostgresEnv)
+    renderPostgresEnv > zdb.env
+  elif [[ $ZITADEL_DATABASE == "cockroach" ]]; then
```
Comment (mlsmaycon) [inline infrastructure_files/getting-started-with-zitadel.sh:474]: this won't catch the case where the user enters ZITADEL_DATABASE=postgres

**Comment by r0b2g1t** [comment]
@mlsmaycon the tests are stable. May you can take a look into it and give feedback.

**Review by mlsmaycon — CHANGES_REQUESTED** [review]

File: infrastructure_files/getting-started-with-zitadel.sh (hunk 7 above)
Comment (mlsmaycon) [reply infrastructure_files/getting-started-with-zitadel.sh:474]: we will handle it after merging.

**Review by mlsmaycon — APPROVED** [review]

**Comment by mlsmaycon** [comment]
@r0b2g1t thanks again for the amazing contribution

**Pull request merged**